src/internet_identity/src/main.rs

```diff
@@ -402,15 +397,12 @@ fn init(maybe_arg: Option<InternetIdentityInit>) {
 fn post_upgrade(maybe_arg: Option<InternetIdentityInit>) {
     init_assets();
     state::init_from_stable_memory();
-    // immediately migrate the persistent state to the new layout
-    state::storage_borrow_mut(|storage| storage.migrate_persistent_state());
+    // load the persistent state after initializing storage, otherwise the memory address to load it from cannot be calculated
+    state::load_persistent_state();
```

Comment:
I moved loading the persistent state upwards, because logically it seems to make more sense. In the future, I'll include it in the `init_from_stable_memory` function to make it a single action.